src/switch_snmp.py

```diff
@@ -0,0 +1,236 @@
+        ['snmpwalk', '-On', '-c', community, '-v2c', host, table_oid],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    rex = re.compile('^([0-9\.]+) = ((.*): )?"?([^"]*)')
+
+    ret_dict = {}
+
+    ret_lines = p.stdout.readlines()
+
+    if (
+        ret_lines[0].find('No Such Object available on this agent at this OID')
+        != -1
+    ):
+        # Would be so much easier to just get snmpwalk return code but
+        # unfortunately it always terminates with code 0
+        return None
+
+    for line in ret_lines:
+        line = line.rstrip()
+
+        # example row:
+        # .1.3.6.1.2.1.31.1.1.1.6.48 = Counter64: 2861909107355
+        rem = rex.match(line)
+        if rem:
+            reg = rem.groups()
+            res_oid = reg[0]
+            res_value = reg[3]
+            res_array = res_oid.split('.')
+            res_num = int(res_array[-1])
+            if skip:
+                if res_num < skip:
+                    ret_dict[res_num] = res_value
+            else:
+                ret_dict[res_num] = res_value
+
+    return ret_dict
+
+
+def get_switch_data(switch, community):
+    graphite = ''
+    switch_model = get_snmp_value(switch, community, '.1.3.6.1.2.1.1.1.0')
+
+    if switch_model is None:
+        return
+
+    skip = None
+
+    if switch_model.find('PowerConnect') != -1:
+        port_name_dict = get_snmp_table(switch, community, port_name_oid, None)
+        switch_type = 'Dell'
+        trunk_group_tab = get_snmp_table(switch, community, cisco_agg_oid, 0)
+
+        trunked_ports = set()
+        active_trunks = set()
+        for port_idx in trunk_group_tab:
+            if int(trunk_group_tab[port_idx]) > 0:
+                trunked_ports.add(int(port_idx))
+                active_trunks.add(int(trunk_group_tab[port_idx]))
+
+    if switch_model.find('ProCurve') != -1:
+        port_name_dict = get_snmp_table(switch, community, port_name_oid, None)
+        switch_type = 'ProCurve'
+        trunk_group_tab = get_snmp_table(switch, community,
+                                         procurve_trunk_oid, 0)
+
+    if switch_model.find('ExtremeXOS') != -1:
+        port_name_dict = get_snmp_table(switch, community,
+                                        port_name_oid, 1000000)
+        switch_type = 'Extreme'
+        trunk_group_tab = None
+        skip = 1000000
+
+    port_num = 0
+    snmp_tables = {}
+    for counter, snmp_oid in COUNTERS.items():
+        snmp_tables[counter] = get_snmp_table(
+            switch, community, snmp_oid, skip
+        )
+
+    # Let's consider that "now" is after all values are
+    # read for this particular switch.
+    now = str(int(time.time()))
+
+    for port_index, port_name in port_name_dict.iteritems():
+        # Skip bad ports. This can be a VLAN on Cisco device.
+        if (
+            port_index not in snmp_tables['bytesIn']
+            or
+            port_index not in snmp_tables['pktsIn']
```

Comment:
I would write something like `not all(port_index in snmp_tables[k] for k in ('bytesIn', 'pktsIn'))`.